### PR TITLE
API: Update Foundatio.Repositories to support System.Text.Json

### PR DESCRIPTION
**Pull request by vlad-kyrychuk-nf**
This solves several serialization issues with system.text.json with the ability to roundtrip polymorphic json, fixed random issues with object to jsonelement and double to int conversions.

### CODE REVIEW

**Review by niemyjski — APPROVED** [review]

File: src/Foundatio.Repositories/Models/Aggregations/IAggregate.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 namespace Foundatio.Repositories.Models;
 
 [JsonConverter(typeof(AggregationsJsonConverter))]
+[System.Text.Json.Serialization.JsonConverter(typeof(AggregationsSystemTextJsonConverter))]
```
Comment (ejsmith) [inline src/Foundatio.Repositories/Models/Aggregations/IAggregate.cs:8]: Maybe we should alias the 2 different `JsonConverter` types like maybe `NSJ.JsonConverter` and `STJ.JsonConverter`?

Comment (ejsmith) [reply src/Foundatio.Repositories/Models/Aggregations/IAggregate.cs:8]: Or maybe just fully qualify both of them.

File: src/Foundatio.Repositories/Models/Aggregations/KeyedBucket.cs [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 
 [DebuggerDisplay("KeyAsString: {KeyAsString} Key: {Key} Total: {Total}")]
 public class KeyedBucket<T> : BucketBase {
     public KeyedBucket() { }
-    public KeyedBucket(IDictionary<string, IAggregate> aggregations) : base(aggregations) { }
 
+    [System.Text.Json.Serialization.JsonConstructor]
+    public KeyedBucket(IReadOnlyDictionary<string, IAggregate> aggregations) : base(aggregations) { }
+
+    // NOTE: object values have been deserializad to json element (https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-converters-how-to?pivots=dotnet-6-0#deserialize-inferred-types-to-object-properties)
```
Comment (ejsmith) [inline src/Foundatio.Repositories/Models/Aggregations/KeyedBucket.cs:14]: I'd rather have this note on the converter class.

**Review by ejsmith — APPROVED** [review]
Some small nits, but overall looks good.

**Pull request merged**